realm/src/main/java/io/realm/DynamicRealmObject.java

```diff
@@ -264,6 +290,86 @@ public boolean hasField(String fieldName) {
+     * to the appropriate input type.
+     * @throws NumberFormatException if a String based number cannot be converted properly.
+     */
+    @SuppressWarnings("unchecked")
+    public void set(String fieldName, Object value) {
+        boolean isString = (value instanceof String);
+        String strValue = isString ? (String) value : null;
+
+        // Do implicit conversion if needed
+        long columnIndex = row.getColumnIndex(fieldName);
+        RealmFieldType type = row.getColumnType(columnIndex);
+        if (isString && type != RealmFieldType.STRING) {
+            switch(type) {
+                case BOOLEAN: value = Boolean.parseBoolean(strValue); break;
+                case INTEGER: value = Long.parseLong(strValue); break;
+                case FLOAT: value = Float.parseFloat(strValue); break;
+                case DOUBLE: value = Double.parseDouble(strValue); break;
+                default:
+                    throw new IllegalArgumentException(String.format("Field %s is not a String field, " +
+                            "and the provide value could not be automatically converted: %s. Use a typed" +
+                            "setter instead", fieldName, value));
+            }
+        }
+
+        if (value == null) {
+            row.nullifyLink(row.getColumnIndex(fieldName));
+            // FIXME Add support for other types when Null is merged.
+        } else {
+            setValue(fieldName, value);
+        }
+    }
+
+    // Automatically finds the appropriate setter based on the objects type
+    private void setValue(String fieldName, Object value) {
+        Class<? extends Object> valueClass = value.getClass();
+        if (valueClass ==  Boolean.class) {
+            setBoolean(fieldName, (Boolean) value);
+        } else if (valueClass == Short.class) {
+            setShort(fieldName, (Short) value);
+        } else if (valueClass == Integer.class) {
+            setInt(fieldName, (Integer) value);
+        } else if (valueClass == Long.class) {
+            setLong(fieldName, (Long) value);
+        } else if (valueClass == Byte.class) {
+            setByte(fieldName, (Byte) value);
+        } else if (valueClass == Float.class) {
+            setFloat(fieldName, (Float) value);
+        } else if (valueClass == Double.class) {
+            setDouble(fieldName, (Double) value);
+        } else if (valueClass == String.class) {
+            setString(fieldName, (String) value);
+        } else if (value instanceof Date) {
+            setDate(fieldName, (Date) value);
+        } else if (value instanceof byte[]) {
+            setBlob(fieldName, (byte[]) value);
+        } else if (valueClass == DynamicRealmObject.class) {
+            setObject(fieldName, (DynamicRealmObject) value);
+        } else if (valueClass == RealmList.class) {
+            // We need to verify that the list only contain DynamicRealmObjects.
```

Comment:
Worth a shot, but no. It cannot tell you the type at Runtime, so all you get out is <E extends RealmObject> which is not useful. I could get the first element and ask for it's type, but then I actually prefer catching the exception as that will only be slow if it is actually wrong.